### PR TITLE
Add validations to book model, clean up specs.

### PR DESCRIPTION
**Pull request by Roshanjossey**
I've Removed some specs that doesn't do anything.
I've Added specs to book model and provided some validations for attributes.
Also, I've used shoulda matchers.

### CODE REVIEW

**Comment by maheshm** [comment]
Please commit from single user so that maintenance is easier.

File: spec/helpers/library_helper_spec.rb [hunk 1]
```diff
@@ -10,6 +10,6 @@
 #     end
 #   end
 # end
-describe LibraryHelper do
-  pending "add some examples to (or delete) #{__FILE__}"
-end
+# describe LibraryHelper do
```
Comment (maheshm) [inline spec/helpers/library_helper_spec.rb]: These are sample codes given by generator as a tip to what you should put here. Please add some useful code here. This need not be a commit as such! :)

**Comment by Roshanjossey** [comment]
@maheshm, I'm sorry that happened. I was using a vagrant box and it did that automatically.
Is there a way to correct it?

File: spec/helpers/library_helper_spec.rb (hunk 1 above)
Comment (Roshanjossey) [inline spec/helpers/library_helper_spec.rb]: I was just trying to get specs pass. (Apparently that was not very easy). Would you recommend rebasing that commit?

Comment (maheshm) [inline spec/helpers/library_helper_spec.rb]: Spec was failing because of #29. Commented code wont fail spec anyway. :)

**Pull request closed without merging**